core-bundle/src/Resources/contao/library/Contao/Widget.php

```diff
@@ -1243,6 +1243,24 @@ public static function getAttributesFromDca($arrData, $strName, $varValue=null,
 	{
 		$arrAttributes = $arrData['eval'] ?? array();
 
+		if (method_exists(System::getContainer(), 'getParameterBag'))
+		{
+			$objParameterBag = System::getContainer()->getParameterBag();
+
+			foreach ($arrAttributes as $strAttrKey => $varAttrValue)
+			{
+				if (!\is_string($varAttrValue) || !preg_match('/%[a-z][a-z0-9_]*\.[a-z0-9_.]+%/i', $varAttrValue, $arrMatches))
+				{
+					continue;
+				}
+
+				$varAttrValue = $objParameterBag->resolveValue($varAttrValue);
```

Comment:
What do you think about adding a `$objParameterBag->has()` check on top of it? Because if you accidentally used `%whatever%` as an `eval` option now, it will resolve to `null` and break your app. With the additional check, this would only happen if this parameter also exists.

Reply:
I tried to make the regular expression in line 1252 very strict to limit this risk. `%whatever%` wouldn’t actually match, but `%whatever.whatever%` would.

We would loose the safety for misspelling parameter names if we only resolve existing parameters. I’m not sure.

Reply:
Okay, we could still add it later. Chances are very limited anyway.